glean-core/python/glean/metrics/timespan.py

```diff
@@ -100,6 +100,37 @@ def cancel(self) -> None:
         def cancel():
             _ffi.lib.glean_timespan_cancel(self._handle)
 
+    class _TimespanContextManager:
+        """
+        A context manager for recording timings. Used by the `measure` method.
+        """
+
+        def __init__(self, timespan: "TimespanMetricType"):
+            self._timespan = timespan
+
+        def __enter__(self) -> None:
+            self._timespan.start()
+
+        def __exit__(self, type, value, tb) -> None:
```

Comment:
I read up on context managers and it seems all 3 values are referring to a potential exception.
Any reason to check tb specifically?
(According to https://docs.python.org/2/library/sys.html#sys.exc_info if there's no exception than all 3 are none, but it doesn't say that `tb` can't be `None` when there _is_ an exception. Then again I don't see how that would happen, so I defer that to you if this is correct)

Reply:
I was just copying the code example in the PEP about context managers: https://docs.python.org/2.5/whatsnew/pep-343.html

But, yeah, maybe you're right that we need to be more careful.  This stuff isn't very well documented -- let me see if I can find a more canonical example.

Reply:
That's good enough for me.